db/template_test.go

```diff
@@ -0,0 +1,201 @@
+// Package db - template tests
+// The following tests validate database functionality, the instance payload.
+// A postgres instance with the tink database schema is required in order to run
+// the following tests.
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/google/uuid"
+	_ "github.com/lib/pq"
+)
+
+// setupCreateTemplate : validates the creation of a new workflow template.
+func setupCreateTemplate(t *testing.T) {
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+
+	id, err := uuid.Parse(ids.templateID)
+	if err != nil {
+		t.Fail()
+	}
+	if _, err := a.tinkDb.instance.Exec("delete from template;"); err != nil {
+		t.Fail()
+	}
+	if err := a.tinkDb.CreateTemplate(a.ctx, "foo", a.templateData, id); err != nil {
+		a.logger.Error(err)
+		t.Fail()
+	}
+}
+
+// TestCreateDuplicateTemplate : creates a duplicate template with the same name.
+// This test should fail. It should only work if the original record is marked
+// for deletion.
+func TestCreateDuplicateTemplate(t *testing.T) {
+	setupCreateTemplate(t)
+	var err error
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+
+	id, err := uuid.NewUUID()
+	if err != nil {
+		t.Fail()
+	}
+	err = a.tinkDb.CreateTemplate(a.ctx, "foo", a.templateData, id)
+	if err != nil {
+		a.logger.Error(err)
+		if !strings.Contains(err.Error(), "exists") {
+			t.Error("function should return: foo already exists")
+		}
+	} else {
+		t.Error("function should return an error")
+	}
+}
+
+// TestCreateNewDuplicateTemplate : creates a duplicate template with the same
+// name, but the previous template will be marked for deletion.
+// This test should pass.
+func TestCreateNewDuplicateTemplate(t *testing.T) {
+	setupInsertIntoDB(t)
+	var err error
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+
+	if err = a.tinkDb.DeleteTemplate(a.ctx, ids.templateID); err != nil {
+		a.logger.Error(err)
+		t.Fail()
+	}
+	id, err := uuid.NewUUID()
+	if err != nil {
+		t.Fail()
+	}
+	if err = a.tinkDb.CreateTemplate(a.ctx, "foo", a.templateData, id); err != nil {
+		a.logger.Error(err)
+		t.Fail()
+	}
+}
+
+// TestUpdatesTemplateOnCreate : updates an existing template if not deleted and
+// the uuid is the same.
+// This test should pass.
+func TestUpdateTemplateOnCreate(t *testing.T) {
+	setupCreateTemplate(t)
+	var err error
+	a := newTestParams(t, ids)
+	defer a.tinkDb.instance.Close()
+
+	id, err := uuid.Parse(ids.templateID)
+	if err != nil {
+		t.Fail()
+	}
+	if err = a.tinkDb.CreateTemplate(a.ctx, "bar", a.templateData, id); err != nil {
+		a.logger.Error(err)
+		t.Fail()
+	}
```

Comment:
this should verify that template was in fact updated and not just that there was no error right?